services/src/main/java/org/keycloak/authentication/actiontoken/updateemail/UpdateEmailActionToken.java

```diff
@@ -29,10 +29,11 @@ public class UpdateEmailActionToken extends DefaultActionToken {
 	@JsonProperty("newEmail")
 	private String newEmail;
 
-	public UpdateEmailActionToken(String userId, int absoluteExpirationInSecs, String oldEmail, String newEmail){
+	public UpdateEmailActionToken(String userId, int absoluteExpirationInSecs, String oldEmail, String newEmail, String clientId){
 		super(userId, TOKEN_TYPE, absoluteExpirationInSecs, null);
 		this.oldEmail = oldEmail;
 		this.newEmail = newEmail;
+		this.issuedFor = clientId;
```

Comment:
Thinking aloud... It is kinda hard to figure out what makes it work because of how the `issuerFor` is set (when serializing/deserializing the token) and how it is used later to establish the client that originated the action.